xc/device.ml

```diff
@@ -608,11 +610,32 @@ let qemu_media_change ~xs device _type params =
 	if is_upstream_qemu device.frontend.domid
 	then begin
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
```

Comment:
If we need to detect QMP event immediately, I think we have to open the QMP socket connection all the time when QEMU is up, put all socket connections in fd_set, use `select` in a thread to detect both `Success`, `Error` and `Event`.

Also need a pipe for each QEMU, because If I write a command to QEMU, I want the response which gets from that thread. So that thread can send to response to me using pipe.